src/core/qgsvectorlayer.cpp

```diff
@@ -4167,6 +4167,11 @@ QVariant QgsVectorLayer::aggregate( QgsAggregateCalculator::Aggregate aggregate,
   if ( ok )
     *ok = false;
 
+  //make a copy or make the param not const
```

Comment:
@nyalldawson if you are fine with this approach I think it would be the best and more robust as opposed to parsing text and hoping that you evaluate/replace the right ones.